backend/dataall/core/resource_threshold/db/resource_threshold_repositories.py

```diff
@@ -0,0 +1,91 @@
+from dataall.core.resource_threshold.db.resource_threshold import ResourceThreshold
+from sqlalchemy import and_
+from datetime import date
+from dataall.base.config import config
+from dataall.base.db import exceptions
+from functools import wraps
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class ResourceThresholdRepository:
+    _RESOURCE_PATHS = {'nlq': 'modules.worksheets.features.max_count_per_day'}
+
+    @staticmethod
+    def _get_count_today(session, username, action_type):
+        amount = (
+            session.query(ResourceThreshold.count)
+            .filter(
+                and_(
+                    ResourceThreshold.username == username,
+                    ResourceThreshold.actionType == action_type,
+                    ResourceThreshold.date == date.today(),
+                )
+            )
+            .scalar()
+        )
+        return amount if amount else 0
+
+    @staticmethod
+    def _add_entry(session, username, action_type):
+        user_entry = ResourceThresholdRepository._get_user_entry(session, username, action_type)
+        if user_entry:
+            session.query(ResourceThreshold).filter(
+                and_(
+                    ResourceThreshold.username == username,
+                    ResourceThreshold.actionType == action_type,
+                )
+            ).update({ResourceThreshold.count: 1, ResourceThreshold.date: date.today()}, synchronize_session=False)
+            session.commit()
+        else:
+            action_entry = ResourceThreshold(username=username, actionType=action_type)
+            session.add(action_entry)
+            session.commit()
+
+    @staticmethod
+    def _increment_count(session, username, action_type):
+        session.query(ResourceThreshold).filter(
+            and_(
+                ResourceThreshold.username == username,
+                ResourceThreshold.actionType == action_type,
+                ResourceThreshold.date == date.today(),
+            )
+        ).update({ResourceThreshold.count: ResourceThreshold.count + 1}, synchronize_session=False)
+        session.commit()
+
+    @staticmethod
+    def _get_user_entry(session, username, action_type):
+        entry = (
+            session.query(ResourceThreshold)
+            .filter(and_(ResourceThreshold.username == username, ResourceThreshold.actionType == action_type))
+            .first()
+        )
+        return entry
+
```

Comment:
check_invocation_count should be part of a service not of the db repository

Reply:
If we create the resoruce_threshold_service all the db functions will not need the prefix `_`

Reply:
I will move the decorator function to a service class